### PR TITLE
Add option to lock InputDisplay to InputRestriction

### PR DESCRIPTION
**Pull request by lambdan**
Adds a option to lock a UINavInputDisplay to a specific EInputRestriction. Useful if you're using InputDisplays to display icons in a options menu, but don't want them to switch out when switching between devices.
Defaults to None, which will follow old behavior (auto detecting).

Tested on version 3.8.6 and Unreal 5.3.2

### CODE REVIEW

File: Source/UINavigation/Public/UINavInputDisplay.h [hunk 1]
```diff
@@ -37,6 +38,10 @@ class UINAVIGATION_API UUINavInputDisplay : public UUserWidget
 	UFUNCTION(BlueprintCallable, Category = "InputDisplay")
 	void SetInputAction(UInputAction* NewAction, const EInputAxis NewAxis, const EAxisType NewScale);
 
+	// Locks to a specific input type. If set to None, it is automatically detected
+	UPROPERTY(BlueprintReadWrite,EditAnywhere,meta=(ExposeOnSpawn),Category="InputDisplay")
+	EInputRestriction InputMethodRestriction;
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavInputDisplay.h:43]: Rename to InputTypeRestriction and set it to None by default here.

File: Source/UINavigation/Private/UINavInputDisplay.cpp [hunk 2]
```diff
@@ -98,9 +98,16 @@ void UUINavInputDisplay::UpdateInputVisuals()
 	if (IsValid(InputRichText)) InputRichText->SetVisibility(ESlateVisibility::Collapsed);
 	InputImage->SetVisibility(ESlateVisibility::Collapsed);
 
+	EInputRestriction Restriction = InputMethodRestriction;
+	if(Restriction == EInputRestriction::None)
+	{
+		// if set to None, detect automatically
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavInputDisplay.cpp:104]: No comment please. I can tell by looking at the code 😁

File: Source/UINavigation/Private/UINavInputDisplay.cpp [hunk 3]
```diff
@@ -98,9 +98,16 @@ void UUINavInputDisplay::UpdateInputVisuals()
...
+		Restriction = UINavPC->IsUsingGamepad() ? EInputRestriction::Gamepad : EInputRestriction::Keyboard_Mouse;
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavInputDisplay.cpp:105]: In addition to this, only bind to the OnInputTypeChanged event in line 31 and 43 is the InputTypeRestriction is None.

File: Source/UINavigation/Public/UINavInputDisplay.h [hunk 4]
```diff
@@ -37,6 +38,10 @@ class UINAVIGATION_API UUINavInputDisplay : public UUserWidget
 	UFUNCTION(BlueprintCallable, Category = "InputDisplay")
 	void SetInputAction(UInputAction* NewAction, const EInputAxis NewAxis, const EAxisType NewScale);
 
+	// Locks to a specific input type. If set to None, it is automatically detected
+	UPROPERTY(BlueprintReadWrite,EditAnywhere,meta=(ExposeOnSpawn),Category="InputDisplay")
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavInputDisplay.h:42]: Add proper spacing here bettwen ",". Also why the expose on spawn?

**Review by goncasmage1 — COMMENTED** [review]
Please merge this into 5.4 instead of master.

File: Source/UINavigation/Public/UINavInputDisplay.h (hunk 4 above)
Comment (lambdan) [reply Source/UINavigation/Public/UINavInputDisplay.h:42]: Thought I needed it for being able to edit in Class Defaults on a child, but I see now that EditAnywhere is enough

**Comment by lambdan** [comment]
Pushed fixes for PR now

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

File: Source/UINavigation/Public/UINavInputDisplay.h [hunk 5]
```diff
@@ -37,6 +38,10 @@ class UINAVIGATION_API UUINavInputDisplay : public UUserWidget
 	UFUNCTION(BlueprintCallable, Category = "InputDisplay")
 	void SetInputAction(UInputAction* NewAction, const EInputAxis NewAxis, const EAxisType NewScale);
 
+	// Locks to a specific input type. If set to None, it is automatically detected
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category="InputDisplay")
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavInputDisplay.h:42]: I don't like that this is BlueprintReadWrite. I'd rather only have this be EditAnywhere and to most BlueprintReadOnly

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

File: Source/UINavigation/Public/UINavInputDisplay.h (hunk 5 above)
Comment (lambdan) [reply Source/UINavigation/Public/UINavInputDisplay.h]: Yeah EditAnywhere is sufficient for our use case too. Removed Blueprint access entirely.

**Review by goncasmage1 — APPROVED** [review]

**Pull request merged**